packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -231,14 +233,30 @@ class Icon extends StatelessWidget {
   /// specified, either directly using this property or using [Directionality].
   final TextDirection? textDirection;
 
+  /// Whether the size is affected by the context's text scaler.
+  ///
+  /// If true, the icon size is going to be scaled by the [TextScaler] of the
+  /// context's [MediaQuery].
+  ///
+  /// Defaults to the nearest [IconTheme]'s
+  /// [IconThemeData.considerTextScale].
+  final bool? considerTextScale;
```

Comment:
Should this just be `bool`?

Reply:
If the passed value is `null`, the value will be obtained through the `context`, while if it's not null the passed value will be used.